app/models/user.rb

```diff
@@ -110,6 +112,18 @@ def validate
   before_validation :dummy_password_for_external_auth
   before_destroy :destroy_subscribed_widget_sets
 
+  def check_reference
+    present_ref = []
+    %w(miq_requests vms miq_widgets miq_templates).each do |association|
```

Comment:
I assume that the user will still have `miq_widgets` since this callback is being prepended and will run before `:destroy_subscribed_widget_sets`, right?

Reply:
yes, this check will be executed first